index.js

```diff
@@ -6,23 +6,45 @@ const { join } = require('node:path');
 const send = require('send');
 const { readFileSync, existsSync } = require('node:fs');
 
+const manifest = new Map();
+
+// Handle route manifest first since it contains the dynamic routes in Next.js v9
+const ROUTE_MANIFEST = '.next/routes-manifest.json';
+
+try {
+	if (existsSync(ROUTE_MANIFEST)) {
+		const routesManifest = JSON.parse(readFileSync(ROUTE_MANIFEST, 'utf8'));
+		
+		for (const route of routesManifest.dynamicRoutes ?? []) {
+			manifest.set(route.page, new RegExp(route.regex));
+		}
+
+		// Later Next.js versions also have static routes in the route manifest
+		for (const route of routesManifest.staticRoutes ?? []) {
+			manifest.set(route.page, new RegExp(route.regex));
+		}
+	}
+} catch (error) {
+	console.error('Could not read routes manifest', error);
+}
+
+// Anything not in the route manifest will be in the pages manifest
 const PATHS_TO_PATHS_FILES = ['.next/serverless/pages-manifest.json', '.next/server/app-paths-routes-manifest.json'];
-let manifestedPaths = [];
+
 for (let path of PATHS_TO_PATHS_FILES) {
 	try {
 		if (!existsSync(path)) continue;
 		const pagesManifest = JSON.parse(readFileSync(path, 'utf8'));
 		for (let key in pagesManifest) {
-			key = key.replace(/\[[^\]]+\]/, (match) => {
```

Comment:
Why remove this part, this is still reading the pages-manifest.json and its parameter syntax, right? Or is it guaranteed that everything with a parameter is in the routes manifest?

Reply:
I don't know if I'd say _guaranteed_ , but that is the idea. I think it may be Next.js version specific too. I'll come up with a better analysis shortly.